src/control/cmd/daos/container.go

```diff
@@ -397,77 +394,51 @@ func (cmd *containerCreateCmd) contCreate() (string, error) {
-	}
-
-	cmd.Infof("Successfully created container %s", contID)
-	return contID, nil
-}
-
-func (cmd *containerCreateCmd) contCreateUNS() (string, error) {
-	var dattr C.struct_duns_attr_t
-
-	props, cleanupProps, err := cmd.getCreateProps()
-	if err != nil {
-		return "", err
+	cleanupContainer := func() {
+		rc := C.daos_cont_destroy(cmd.cPoolHandle, cContID, goBool2int(true), nil)
+		if err := daosError(rc); err != nil {
+			cmd.Noticef("Failed to clean-up container %v", err)
+		}
 	}
-	defer cleanupProps()
-	dattr.da_props = props
 
-	if !cmd.Type.Set {
-		return "", errors.New("container type is required for UNS")
-	}
-	dattr.da_type = cmd.Type.Type
+	if len(cmd.Attrs.ParsedProps) != 0 {
+		attrs := make(attrList, 0, len(cmd.Attrs.ParsedProps))
+		for key, val := range cmd.Attrs.ParsedProps {
+			attrs = append(attrs, &attribute{
+				Name:  key,
+				Value: []byte(val),
+			})
+		}
 
-	if cmd.poolUUID != uuid.Nil {
-		poolUUIDStr := C.CString(cmd.poolUUID.String())
-		defer freeString(poolUUIDStr)
-		C.uuid_parse(poolUUIDStr, &dattr.da_puuid[0])
-	}
-	if cmd.contUUID != uuid.Nil {
-		contUUIDStr := C.CString(cmd.contUUID.String())
-		defer freeString(contUUIDStr)
-		C.uuid_parse(contUUIDStr, &dattr.da_cuuid[0])
-	}
+		if err := cmd.openContainer(C.DAOS_COO_RW); err != nil {
+			cleanupContainer()
+			return "", errors.Wrapf(err, "failed to open new container %s", contID)
+		}
+		defer cmd.closeContainer()
 
-	if cmd.ChunkSize.Set {
-		dattr.da_chunk_size = cmd.ChunkSize.Size
-	}
-	if cmd.ObjectClass.Set {
-		dattr.da_oclass_id = cmd.ObjectClass.Class
-	}
-	if cmd.DirObjectClass.Set {
-		dattr.da_dir_oclass_id = cmd.DirObjectClass.Class
-	}
-	if cmd.FileObjectClass.Set {
-		dattr.da_file_oclass_id = cmd.FileObjectClass.Class
-	}
-	if cmd.CHints != "" {
-		hint := C.CString(cmd.CHints)
-		defer freeString(hint)
-		C.strncpy(&dattr.da_hints[0], hint, C.DAOS_CONT_HINT_MAX_LEN-1)
+		if err := setDaosAttributes(cmd.cContHandle, contAttr, attrs); err != nil {
+			cleanupContainer()
+			return "", errors.Wrapf(err, "failed to set user attributes on new container %s", contID)
+		}
 	}
 
-	cPath := C.CString(cmd.Path)
-	defer freeString(cPath)
+	if cmd.Path != "" {
+		cPath := C.CString(cmd.Path)
+		defer freeString(cPath)
 
-	dunsErrno := C.duns_create_path(cmd.cPoolHandle, cPath, &dattr)
-	rc := C.daos_errno2der(dunsErrno)
-	if err := daosError(rc); err != nil {
-		return "", errors.Wrapf(err, "duns_create_path() failed")
+		dunsErrno := C.duns_link_cont(cmd.cPoolHandle, cContID, cPath)
+		rc := C.daos_errno2der(dunsErrno)
+		if err := daosError(rc); err != nil {
+			cleanupContainer()
+			return "", errors.Wrapf(err, "duns_link_cont() failed")
+		}
 	}
 
-	contID := C.GoString(&dattr.da_cont[0])
-	cmd.contUUID, err = uuid.Parse(contID)
-	if err != nil {
-		cmd.contLabel = contID
-	}
-	cmd.Infof("Successfully created container %s type %s", contID, cmd.Type.String())
+	cmd.Infof("Successfully created container %s", contID)
```

Comment:
Curious - Was the `type %s` intentionally removed? I think it was previously only printed for POSIX, but now it never is.

Reply:
Just keep the message of the main merged function.
However, I also prefer this log message as it contains more relevant infos.
Thus I will change it.

- [ ] Add container type in the log message

Reply:
> Just keep the message of the main merged function. However, I also prefer this log message as it contains more relevant infos. Thus I will change it.
> 
> * [x]  Add container type in the log message

Fixed with commit 7f267334e